src/restore_config.c

```diff
@@ -603,6 +618,16 @@ restore_config_init(int argc, char* argv[], restore_config_t* conf)
 		return RESTORE_CONFIG_INIT_FAILURE;
 	}
 
+	if (conf->directory && conf->directory_list) {
+		err("Invalid options: --directory and --directory-list are mutually exclusive.");
+		return RESTORE_CONFIG_INIT_FAILURE;
+	}
+
+	if (conf->input_file && conf->directory_list) {
+		err("Invalid options: --input-file and --directory-list are mutually exclusive.");
+		return RESTORE_CONFIG_INIT_FAILURE;
+	}
+
```

Comment:
What about `--input-file` and `--directory`? Is that invalid?

Reply:
Yes, there is a check for that on line 602.

Reply:
Maybe we could move that check closer to the other mutual exclusion checks. Should these checks also use `!=  NULL` like the others?

Reply:
makes sense, done